benchmarks/timeseries/periodic_feature_extractor.py

```diff
@@ -0,0 +1,213 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""
+Benchmarks for the PeriodicFeatureExtractor class.
+
+This module contains benchmarks that are comparing
+the performance of a numpy implementation with a python
+implementation.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from datetime import datetime, timedelta, timezone
+from functools import partial
+from timeit import timeit
+from typing import List
+
+import numpy as np
+from frequenz.channels import Broadcast
+from numpy.random import default_rng
+from numpy.typing import NDArray
+
+from frequenz.sdk.timeseries import Sample
+from frequenz.sdk.timeseries._moving_window import MovingWindow
+from frequenz.sdk.timeseries.periodic_feature_extractor import PeriodicFeatureExtractor
+
+
+async def init_feature_extractor(period: int) -> PeriodicFeatureExtractor:
+    """Initialize the PeriodicFeatureExtractor class."""
+    # We only need the moving window to initialize the PeriodicFeatureExtractor class.
+    lm_chan = Broadcast[Sample]("lm_net_power")
+    # monkey patch the MovingWindow class to have a fixed size such that
+    # it can be initialized with a period bigger then the window size.
+    MovingWindow.__len__ = lambda self: period  # type: ignore[method-assign]
+    moving_window = MovingWindow(
+        timedelta(seconds=1), lm_chan.new_receiver(), timedelta(seconds=1)
+    )
+
+    await lm_chan.new_sender().send(Sample(datetime.now(tz=timezone.utc), 0))
+
+    # Initialize the PeriodicFeatureExtractor class with a period of period seconds.
+    # This works since the sampling period is set to 1 second.
+    return PeriodicFeatureExtractor(moving_window, timedelta(seconds=period))
+
+
+def _calculate_avg_window(
+    feature_extractor: PeriodicFeatureExtractor,
+    window: NDArray[np.float_],
+    window_size: int,
+) -> NDArray[np.float_]:
+    """
+    Reshapes the window and calculates the average.
+
+    This method calculates the average of a window by averaging over
+    all windows fully inside the passed numpy array having the distance
+    `self.distance`.
+
+    Args:
+        feature_extractor: The instance of the PeriodicFeatureExtractor to use.
+        window: The window to calculate the average over.
+        window_size: The size of the window to calculate the average over.
+        weights: The weights to use for the average calculation.
+
+    Returns:
+        The averaged window.
+    """
+    reshaped = feature_extractor._reshape_np_array(  # pylint: disable=W0212
```

Comment:
Can you maybe replace the `pylint` numeric codes with the text representation (in this case `W0212` -> `protected-access`)? It is much harder to know what's being ignored otherwise.